crates/daemon/src/wallet.rs

```diff
@@ -158,10 +154,61 @@ impl Actor<ElectrumBlockchain, sled::Tree> {
 
         Ok((addr, receiver))
     }
+
+    fn build_wallet(ext_priv_key: ExtendedPrivKey, db: Db) -> Result<Wallet<Tree>> {
+        let wallet_name = wallet_name_from_descriptor(
+            bdk::template::Bip84(ext_priv_key, KeychainKind::External),
+            Some(bdk::template::Bip84(ext_priv_key, KeychainKind::Internal)),
+            ext_priv_key.network,
+            &Secp256k1::new(),
+        )?;
+
+        let db = db.open_tree(wallet_name)?;
+
+        let wallet = Wallet::new(
+            bdk::template::Bip84(ext_priv_key, KeychainKind::External),
+            Some(bdk::template::Bip84(ext_priv_key, KeychainKind::Internal)),
+            ext_priv_key.network,
+            db,
+        )?;
+
+        Ok(wallet)
+    }
+}
+
+#[xtra_productivity]
+impl<B> Actor<B, Tree>
+where
+    Self: xtra::Actor,
+{
+    pub fn import_seed(&mut self, msg: ImportSeed) -> Result<AddressInfo> {
+        let seed = msg.seed;
+        let import_seed: [u8; 256] = seed
+            .clone()
+            .try_into()
+            .map_err(|_| anyhow!("seed must be 256 bytes long"))?;
```

Comment:
🔧 This is more idiomatic.

```suggestion
            // use anyhow::Context;
            .context("seed must be 256 bytes long")?;
```